packages/shared/components/popups/MintNftFormPopup.svelte

```diff
@@ -0,0 +1,158 @@
+<script lang="typescript">
+    import { onMount } from 'svelte'
+    import { BaseError } from '@core/error'
+    import { localize } from '@core/i18n'
+    import { setMintNftDetails, mintNftDetails, MimeType, TokenStandard } from '@core/wallet'
+    import { handleError } from '@core/error/handlers/handleError'
+    import { closePopup, openPopup } from '@lib/popup'
+    import { Button, Dropdown, Error, FontWeight, OptionalInput, Text, TextInput } from 'shared/components'
+
+    export let _onMount: (..._: any[]) => Promise<void> = async () => {}
+
+    let { type, uri, name, collectionId, collectionName, royalties, issuerName, description, attributes } =
+        $mintNftDetails
+
+    let nameError: string = ''
+    $: name, (nameError = '')
+    let uriError: string = ''
+    $: uri, (uriError = '')
+    let typeError: string = ''
+    $: type, (typeError = '')
+
+    let error: BaseError
+
+    const nftTypeOptions: { label: MimeType; value: MimeType }[] = [
+        { label: 'image/jpeg', value: 'image/jpeg' },
+        { label: 'image/png', value: 'image/png' },
+        { label: 'image/gif', value: 'image/gif' },
+        { label: 'application/pdf', value: 'application/pdf' },
+        { label: 'text/plain', value: 'text/plain' },
+    ]
+
+    function handleCancel() {
+        closePopup()
+    }
+
+    function handleContinue() {
+        const valid = validate()
+        if (valid) {
+            setMintNftDetails({
+                id: undefined,
+                standard: TokenStandard.IRC27,
+                version: undefined,
+                type,
+                uri,
+                name,
+                collectionId,
+                collectionName,
+                royalties,
+                issuerName,
+                description,
+                attributes,
+            })
+            openPopup({
+                type: 'mintNftConfirmation',
+                overflow: true,
+            })
+        }
+    }
+
+    function handleSelectNftType(item: { label: MimeType; value: MimeType }) {
+        type = item.value
+    }
+
+    function validate(): boolean {
+        // TODO: implement validation
```

Comment:
do we want to include that in this PR?

Reply:
I suggest we make a different PR for this and address the royalties and attributes fields as well.